### PR TITLE
Lint Python code with ruff

### PR DESCRIPTION
**Pull request by cclauss**
[Ruff](https://beta.ruff.rs/) supports [over 500 lint rules](https://beta.ruff.rs/docs/rules) including bandit, isort, pylint, pyupgrade, and flake8 plus its plugins, and is written in Rust for speed.

The `ruff` Action uses minimal steps to run in ~10 seconds, rapidly providing intuitive GitHub Annotations to contributors.

image

Moves `isort` out of `tox.ini` and into `ruff` and `pyproject.toml`.
Now when you want to fix imports, just do `ruff --select="I" --fix.`

### CODE REVIEW

**Comment by melroy89** [comment]
OK. But I'm already running black. Black is also an auto formatter / linter.

Will this not interfere with Black?

**Comment by cclauss** [comment]
https://beta.ruff.rs/docs/faq/#is-ruff-compatible-with-black
TL;DR: Black is a formatter and Ruff is a linter that has been designed to be compatible with Black.

File: pyproject.toml [hunk 1]
```diff
@@ -47,6 +47,27 @@ content-type = "text/markdown"
 [project.urls]
 Homepage = "https://github.com/fake-useragent/fake-useragent"
 
+[tool.ruff]
+line-length = 142
+select = ["B", "C4", "C9", "E", "F", "I", "PL", "S", "SIM", "U", "W", "YTT"]
+ignore = ["B904", "C408", "PLW2901", "SIM105", "SIM108"]
+target-version = "py37"
```
Comment (melroy89) [inline pyproject.toml:54]: Since `isort` isn't supported anymore for Python 3.7. Should we move to Python 3.8 here as well maybe?

**Review by melroy89 — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 1 above)
Comment (melroy89) [reply pyproject.toml:54]: (also maybe rebase, just to be sure)

Comment (cclauss) [reply pyproject.toml:54]: Rebased.

Py37 still has several months before EOL -- https://devguide.python.org/versions/#versions

Comment (melroy89) [reply pyproject.toml:54]: hm ok. Since 3.7 was giving issues during GitHub Actions, I removed it from the workflow and tox.

**Review by melroy89 — APPROVED** [review]
